src/cpp/FilesInformation/Loaders/FilesByNameLoader.cpp

```diff
@@ -109,17 +109,33 @@ TEST_CASE("Non Existing file") {
     delete file;
 }
 
+TEST_CASE("Non Existing directory") {
+    FilesByNameLoader loader;
+    Directory* directory = loader.loadDirectory("tmp");
+    CHECK(directory == nullptr);
+    delete directory;
+}
 
 TEST_CASE("Empty directory") {
+    std::filesystem::create_directory("tmp");
+
     FilesByNameLoader loader;
-    Directory* directory = loader.loadDirectory("resources/empty_directory_test");
+    Directory* directory = loader.loadDirectory("tmp");
     CHECK(directory->size() == 0);
     delete directory;
+
+    std::filesystem::remove("tmp");
 }
 
 TEST_CASE("Directory with files") {
```

Comment:
Wrong name? This is directory with another nested directory, not directory with files
Also you may want to check that Directory object returned contains expected items, not just that it is not empty

Reply:
And also a check for a directory with files may be needed too